backend/dissemination/search.py

```diff
@@ -12,46 +12,30 @@ def search_general(
     cog_or_oversight=None,
     agency_name=None,
     audit_years=None,
+    include_private=False,
 ):
-    query = Q(is_public=True)
+    query = Q()
 
-    if alns:
-        query.add(_get_aln_match_query(alns), Q.AND)
-
-    if names:
-        query.add(_get_names_match_query(names), Q.AND)
-
-    if uei_or_eins:
-        uei_or_ein_match = Q(
-            Q(auditee_uei__in=uei_or_eins) | Q(auditee_ein__in=uei_or_eins)
-        )
-        query.add(uei_or_ein_match, Q.AND)
-
-    if start_date:
-        start_date_match = Q(fac_accepted_date__gte=start_date)
-        query.add(start_date_match, Q.AND)
-
-    if end_date:
-        end_date_match = Q(fac_accepted_date__lte=end_date)
-        query.add(end_date_match, Q.AND)
-
-    if cog_or_oversight:
-        if cog_or_oversight.lower() == "cog":
-            cog_match = Q(cognizant_agency__in=[agency_name])
-            query.add(cog_match, Q.AND)
-        elif cog_or_oversight.lower() == "oversight":
-            oversight_match = Q(oversight_agency__in=[agency_name])
-            query.add(oversight_match, Q.AND)
-
-    if audit_years:
-        fiscal_year_match = Q(audit_year__in=audit_years)
-        query.add(fiscal_year_match, Q.AND)
+    query.add(_get_is_public_query(include_private), Q.AND)
+    query.add(_get_aln_match_query(alns), Q.AND)
+    query.add(_get_names_match_query(names), Q.AND)
+    query.add(_get_uei_or_eins_match_query(uei_or_eins), Q.AND)
+    query.add(_get_start_date_match_query(start_date), Q.AND)
+    query.add(_get_end_date_match_query(end_date), Q.AND)
+    query.add(_get_cog_or_oversight_match_query(agency_name, cog_or_oversight), Q.AND)
+    query.add(_get_audit_years_match_query(audit_years), Q.AND)
 
     results = General.objects.filter(query).order_by("-fac_accepted_date")
 
     return results
 
 
+def _get_is_public_query(include_private):
```

Comment:
Would it make more sense to have an `if` clause around this, rather than having this clash between the function name and the variable name?

Reply:
👍 moved the `is_public` query clause up to the root function for clarity